tests/bears/DebugBearsTest.py

```diff
@@ -0,0 +1,68 @@
+import unittest
+import sys
+
+from io import StringIO
+
+from coalib.bears.Bear import Debugger, debug_run
+
+
+def func1(*args, **kwargs):
+    yield 1
+    yield 2
+    yield 3
+
+
+def func2(*args, **kwargs):
+    return([1, 2])
+
+
+def func3(*args, **kwargs):
+    return func1(*args, **kwargs)
+
+
+def execute_debugger(debugger_commands, func, *args, **kwargs):
+    input = StringIO('\n'.join(debugger_commands))
+    output = StringIO()
+    dbg = Debugger(stdin=input, stdout=output)
+    return debug_run(func, dbg, *args, **kwargs), output.getvalue()
+
+
+class DebugBearsTest(unittest.TestCase):
+    def setUp(self):
+        # restore the coverage settrace to prevent the coverage breakage
+        # on project because  we can't chain coverage trace to run parallel
+        # with debugger. To increase the coverage Mock test has been added in
+        # BearTest file.
+        # https://goo.gl/sKaJfh
+        self.trace = sys.gettrace()
+
+    def tearDown(self):
+        sys.settrace(self.trace)
+
+    def test_run_return_yield_with_debugger(self):
+        args = ('a', ('b', 'c'))
+        kwargs = {'d': 'e'}
+        result, output = execute_debugger('qcqc', func1, *args, **kwargs)
+        self.assertEqual(result, [1, 2, 3])
+        lines = output.splitlines()
+        self.assertEqual(lines[1], '-> yield 1')
+        self.assertEqual(lines[3], '-> yield 2')
+        self.assertEqual(lines[5], '-> yield 3')
+
+    def test_run_return_list_with_debugger(self):
+        args = ('a', ('b', 'c'))
+        kwargs = {'d': 'e'}
+        result, output = execute_debugger('qc', func2, *args, **kwargs)
+        self.assertEqual(result, [1, 2])
+        lines = output.splitlines()
+        self.assertEqual(lines[1], '-> return([1, 2])')
+
+    def test_run_return_generator_with_debugger(self):
+        args = ('a', ('b', 'c'))
+        kwargs = {'d': 'e'}
+        result, output = execute_debugger('qcqcqc', func3, *args, **kwargs)
```

Comment:
shouldn't it be just 4 commands to make the debugger quit completely? Please check that up, it's good to use only as many commands as necessary, to catch possible bugs (debugger runs further actually and displays more lines of code, but we don't check additional lines)

Reply:
Same for the other tests above, please revise the number of commands necessary to let the debugger finish completely  :)

Reply:
actually, with generators, it will need one more so that in that it will throw `StopIterationError` and after that It will return a list (if I don't add extra command it will return something like `[1,2,3,None]`
without even catching the  `StopIterationError`

Reply:
Makes sense, nvm then :+1: